### PR TITLE
add documentation for the jarrett api function

### PR DESCRIPTION
**Pull request by inoryy**
I've noticed that it's used in `jax.nn`, seems to be generally useful transformation, but naming is a bit unintuitive. I've taken the example from #525.

### CODE REVIEW

File: jax/api.py [hunk 1]
```diff
@@ -1707,6 +1707,47 @@ def primal_fun(*args, **kwargs):
...
+  >>> def f(x):
+  ...   return jax.numpy.sin(jax.numpy.sin(jax.numpy.sin(x)))
+  ...
+  >>> f2 = jax.jarrett(f)
+  >>> (_, f_vjp), (_, f2_vjp) = jax.vjp(f, 3.), jax.vjp(f2, 3.)
+  >>> print f_vjp(1.), f2_vjp(1.)
+  (array(-0.9704719, dtype=float32),), (array(-0.9704719, dtype=float32),)
+  >>> print make_jaxpr(f_vjp)(1.)
+  { lambda e i g ;  ; a.
+    let b = pack * a
```
Comment (jekbradbury) [inline jax/api.py]: These jaxprs appear to be outdated (they contain `pack`, which was removed in detuplification). Can you regenerate?

**Comment by hawkinsp** [comment]
Are we comfortable freezing this as the API (especially the name)?

File: jax/api.py [hunk 2]
```diff
@@ -1707,6 +1707,47 @@ def primal_fun(*args, **kwargs):
...
+  For example:
+
+  >>> def f(x):
+  ...   return jax.numpy.sin(jax.numpy.sin(jax.numpy.sin(x)))
+  ...
+  >>> f2 = jax.jarrett(f)
+  >>> (_, f_vjp), (_, f2_vjp) = jax.vjp(f, 3.), jax.vjp(f2, 3.)
+  >>> print f_vjp(1.), f2_vjp(1.)
+  (array(-0.9704719, dtype=float32),), (array(-0.9704719, dtype=float32),)
+  >>> print make_jaxpr(f_vjp)(1.)
```
Comment (shoyer) [inline jax/api.py]: Let’s try to stick with Python 3 compatible syntax for all examples.

**Comment by hawkinsp** [comment]
We've removed `jarrett` because of bitrot, so this PR is moot at the moment. (When we revive it, we should include this doc string!)

**Pull request closed without merging**